java/opendcs/src/main/resources/db/CWMS-Oracle/callbacks/afterMigrate__activateOfficeTrigger.sql

```diff
@@ -0,0 +1,18 @@
+declare
+    the_sql varchar2(4000);
+begin
+
+  for rec in (
+    select table_name from user_tab_columns where column_name in ('DB_OFFICE_CODE')
+    )
+  loop
+    execute immediate 'alter table ' || rec.table_name || ' modify db_office_code default null';
+    the_sql := 
+    'create or replace trigger ' || rec.table_name || '_get_office' ||
+    ' before insert or update on ' || rec.table_name ||
+    ' for each row begin set_office_column(:new.db_office_code,:old.db_office_code); end;';
```

Comment:
Will that work --  passing :new.db_office_code  by 'reference'  into set_office_column?  The :new.db_office_code feels like it is a value type.

Reply:
`:new` and `:old` are triggerisms. They are provided by the engine that executes the trigger.

one or the other may be null, otherwise they will match the datatype of the insert itself. the set_office_column method then handles the various conditions; which due to how the triggers work I was not able to handle in the trigger directly. (There can be a lot of quirks in triggers.)